### DISCUSSION TITLE
Select text between matching pairs

### DISCUSSION THREAD

**Discussion by RoastBeefer00** [helix-editor/helix#8789] [General]
Let's say I have this code:
```
name = "name"
word = "word"
```

If I'm at the beginning of the second line and type `mi"` it selects the quotation mark at the end of the previous line to the first quotation on line 2. I can't think why I'd ever want this, it's pretty obvious I'd like to select the text between the quotes on line 2. `f"lmi"` seems to be the fastest way to accomplish what I want.

My suggestion is to either prioritize the current line or create a setting to have that preference. 

For example typing `vi"` in Vim will select all text between quotes on the current line and is significantly faster and more intuitive than what Helix currently offers.

This same concept applies to many other things like brackets, parentheses, etc.

**Comment by pascalkuthe** [comment]
This is working as intended, unlike vim mi" does not jump forward to the next pair (if you use it at the start of the first line it will just do nothing for example).


This behavior is intentional and we don't plan to change it.

**Reply by RoastBeefer00** [reply]
May I ask the reasoning behind this? I can't imagine why this would be considered a feature or why I wouldn't want that functionality.

**Reply by the-mikedavis** [reply]
Vim's jumping behavior makes it difficult to predict what the movement will do especially once you have multiple cursors. We had the jumping behavior before for increment/decrement (`C-a`/`C-x`) for example and intentionally removed it because it was more often hurting than helping - you would jump to numbers far later in the line/file than you expected.

**Reply by jakubkaczor** [reply]
I don't think the example with incrementing/decrementing is relevant here. It doesn't involve the concept of _pairs_. I believe the confusion of the author is caused by thinking the “m” command is for matching _pairs_ (see the title) which is not the case in the only meaningful sense I can think of. For example, in the given example, the `"` characters surrounding selection are not matching pairs in the sense of the language they were written in. I don't know a language in which they would be. Therefore, I think the current behavior is confusing, and it should jump to the closest pair minimizing some metric. If thought otherwise, could you provide an example where the current behavior is more expected?

I far often use the discussed command with a single cursor. If for some reason you consider it problematic with multiple, I think having the current behavior on some command, and the requested one mapped to a key sequence would hurt less, then delegating the requested one to some other command. This is under assumption that mapping the requested behavior to the key sequence would pollute key maps set.

**Comment by samrocksc** [comment]
this answer/discussion attitude turns me off from this editor adoption :(

**Reply by kirawi** [reply]
This is being worked on in https://github.com/helix-editor/helix/pull/11260
There might be some misunderstanding of what the discussion is about. This discussion was asking for changing the behavior of the existing commands, but we were open to an additional command.
